src/ncp/ncp_spi.cpp

```diff
@@ -321,16 +320,27 @@ ThreadError NcpSpi::PrepareNextSpiSendFrame(void)
     return errorCode;
 }
 
-void NcpSpi::HandleSendDone(void *aContext)
+void NcpSpi::PrepareTxFrame(void *aContext)
 {
-    static_cast<NcpSpi*>(aContext)->HandleSendDone();
+    static_cast<NcpSpi*>(aContext)->PrepareTxFrame();
 }
 
-void NcpSpi::HandleSendDone(void)
+void NcpSpi::PrepareTxFrame(void)
 {
-    mSending = false;
-    PrepareNextSpiSendFrame();
-    mHandlingSendDone = false;
+    if (mHandlingSendDone)
+    {
+        mSending = false;
+    }
+
+    if (!mSending)
+    {
+        PrepareNextSpiSendFrame();
+    }
+
+    if (mHandlingSendDone)
```

Comment:
Since `mHandlingSendoDone`can be set from ISR context, we should clear it only if it was set when we entered this method. I fixed this and simplified the implementation in new commit.